### PR TITLE
ADEN-7955 UAP CTP tests and refactor

### PR DESCRIPTION
**Pull request by emadry**
UAP CTP Tests added
Refactor
Change of resolution to match linux runner

### CODE REVIEW

File: tests/common/helpers.js [hunk 1]
```diff
@@ -325,9 +329,26 @@ class Helpers {
 		browser.frame(frame);
 	}
 
-	setWindowSize(width = 1920, height = 1080) {
+	/**
+	 * Set window size to full HD.
+	 * @param width
+	 * @param height
+	 */
+	setWindowSize(width = 1600, height = 900) {
```
Comment (dmnsobczak) [inline tests/common/helpers.js]: That's not FullHD resolution and you call it without any parameters. Is it correct?

File: tests/common/helpers.js [hunk 2]
```diff
@@ -325,9 +329,26 @@ class Helpers {
...
 		browser.windowHandleSize({ width, height });
 	}
+
+	/**
+	 * Takes slot size and its ratio and waits for the desired dimensions.
+	 * @param adSlot Slot to take dimensions from
+	 * @param ratio value to divide by
+	 */
+	waitForResolved(adSlot, ratio) {
+		browser.waitUntil(() => browser.getElementSize(adSlot, 'height') >= (this.calculateHeightWithRatio(adSlot, ratio) - comparisonOffsetPx),
```
Comment (dmnsobczak) [inline tests/common/helpers.js]: I think that AdEngine lint would expect to wrap part from `browser.get...` to `...OffsetPx)` in parenthesis, but I might be wrong.

File: tests/common/helpers.js [hunk 3]
```diff
@@ -325,9 +329,26 @@ class Helpers {
...
+			timeouts.standard,
+			'Dimensions not changed',
+			timeouts.interval);
```
Comment (dmnsobczak) [inline tests/common/helpers.js]: New line on `(` and `);` - I think it's increasing readability.

File: tests/specs/templates/hivi-uap-ad.desktop.test.js [hunk 4]
```diff
@@ -185,23 +186,28 @@ describe('Desktop HiVi UAP ads page: bottom leaderboard', () => {
 	let videoFinishedDimensions;
 
 	before(() => {
-		helpers.reloadPageAndWaitForSlot(hiviUap.pageLink, adSlots.topLeaderboard);
+		// helpers.reloadPageAndWaitForSlot(adSlots.topLeaderboard);
```
Comment (dmnsobczak) [inline tests/specs/templates/hivi-uap-ad.desktop.test.js]: What's a purpose of leaving it here?

File: tests/specs/templates/hivi-uap-ctp-ad.desktop.test.js [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+
+describe('Desktop HiVi UAP CTP ads page: bottom leaderboard', () => {
+	let adStatus;
+	let defaultDimensions;
+	let refreshDimensions;
+	let videoFinishedDimensions;
+
+	before(() => {
+		hiviUapCtp.openUapWithState(false, hiviUapCtp.pageLink, adSlots.topLeaderboard);
+		// helpers.reloadPageAndWaitForSlot(adSlots.topLeaderboard);
```
Comment (dmnsobczak) [inline tests/specs/templates/hivi-uap-ctp-ad.desktop.test.js]: Please add comment why it should be left here.

File: wdio.desktop.conf.js [hunk 6]
```diff
@@ -4,7 +4,10 @@
 const merge = require('deepmerge');
 const wdioConf = require('./wdio.conf.js');
 
+global.wdioEnvironment = 'desktop';
+
 exports.config = merge(wdioConf.config, {
+
```
Comment (dmnsobczak) [inline wdio.desktop.conf.js]: Not needed.

**Comment by dmnsobczak** [comment]
LGTM 👍 Minor comments left, but they're optional.

File: tests/specs/templates/hivi-uap-ctp-ad.desktop.test.js (hunk 5 above)
Comment (emadry) [reply tests/specs/templates/hivi-uap-ctp-ad.desktop.test.js]: It shouldn't

File: tests/specs/templates/hivi-uap-ad.desktop.test.js (hunk 4 above)
Comment (emadry) [reply tests/specs/templates/hivi-uap-ad.desktop.test.js]: There's none

**Pull request merged**